pkg/operator/apis/monitoring/v1/scrape_config.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"net/url"
+
+	"github.com/alecthomas/units"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	promconfig "github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/model/relabel"
+)
+
+type Config struct {
```

Comment:
Some sustainability questions:

1. We desperately need commentary on why we are vendoring this.
2. How to track what fields we removed? What's the process on when we are upgrading Prometheus and more fields needs to be removed or added?

Reply:
I think Prometheus would not accept omitzero even in the latest version, because we never envisioned our structs to be Marshalled (only unmarshalled). Thanks!